test/test_suit/test_snowpipe_streaming_nullable_values_after_smt.py

```diff
@@ -0,0 +1,54 @@
+import json
+from snowflake.connector import DictCursor
+from test_suit.test_utils import NonRetryableError, RetryableError
+
+
+# Testing behavior for behavior.on.null.values = IGNORE and SMTs that can return null values.
+# Schematization is disabled.
+class TestSnowpipeStreamingNullableValuesAfterSmt:
+
+    def __init__(self, driver, nameSalt):
+        self.driver = driver
+        self.fileName = 'snowpipe_streaming_nullable_values_after_smt'
+        self.table = self.fileName + nameSalt
+        self.topic = self.table
+
+    def getConfigFileName(self):
+        return self.fileName + '.json'
+
+    def send(self):
+        value = []
+        for idx in range(200):
+            event = { 'index': idx, 'someKey': 'someValue' }
+
+            if idx % 2 == 0: # Only every other event contains optionalField.
+                optional_value = { 'index': idx, 'from_optional_field': True }
+                event.update({ 'optionalField': optional_value })
+
+            value.append(json.dumps(event).encode('utf-8'))
+
+        self.driver.sendBytesData(self.topic, value)
+
+    def verify(self, round):
+        cur = self.driver.snowflake_conn.cursor(DictCursor)
+        res = cur.execute(f'select record_content, record_metadata:offset::number as offset from {self.table}').fetchall()
+
+        if len(res) == 0:
+            raise RetryableError()
```

Comment:
Will retrying be eventually stopped if anything goes wrong here? After how many attempts?

Reply:
There are [30 retries](https://github.com/snowflakedb/snowflake-kafka-connector/blob/master/test/test_verify.py#L50) before failure.